### PR TITLE
Allow method chaining in logger classes

### PR DESCRIPTION
**Pull request by serhiikobalt**
Return reference to itself instead of void in setters

Resolves: OLPEDGE-403
Signed-off-by: Serhii Lysenko <ext-serhii.lysenko@here.com>

### CODE REVIEW

File: olp-cpp-sdk-core/include/olp/core/logging/Configuration.h [hunk 1]
```diff
@@ -87,27 +87,28 @@ class CORE_API Configuration {
    * added appender is set to logging::Level::Trace.
    * @param formatter The message formatter.
    */
-  void addConsoleAppender(
+  Configuration& addConsoleAppender(
```
Comment (andescu) [inline olp-cpp-sdk-core/include/olp/core/logging/Configuration.h]: addConsoleAppender() and addDebugAppender() methods should disapear anyway

File: olp-cpp-sdk-core/include/olp/core/logging/Configuration.h [hunk 2]
```diff
@@ -131,13 +132,17 @@ inline Configuration& Configuration::operator=(Configuration&& other) noexcept {
 
 inline bool Configuration::isValid() const { return !m_appenders.empty(); }
```
Comment (andescu) [inline olp-cpp-sdk-core/include/olp/core/logging/Configuration.h]: move all below methods to separate Configuration.inl file

Comment (andescu) [reply olp-cpp-sdk-core/include/olp/core/logging/Configuration.h]: inline Configuration::Configuration() {} can be removed mark and replaced by:
```cpp
Configuration() = default;
```

File: olp-cpp-sdk-core/include/olp/core/logging/Configuration.h [hunk 3]
```diff
@@ -131,13 +132,17 @@ inline Configuration& Configuration::operator=(Configuration&& other) noexcept {
...
 
-inline void Configuration::addAppender(std::shared_ptr<IAppender> appender,
-                                       logging::Level level) {
+inline Configuration& Configuration::addAppender(
+    std::shared_ptr<IAppender> appender, logging::Level level) {
   AppenderWithLogLevel appenderWithLogLevel = {level, std::move(appender)};
   m_appenders.emplace_back(std::move(appenderWithLogLevel));
```
Comment (andescu) [inline olp-cpp-sdk-core/include/olp/core/logging/Configuration.h]: Should work also:
```cpp
m_appenders.emplace_back({level, std::move(appender)});
```

File: olp-cpp-sdk-core/include/olp/core/logging/Configuration.h [hunk 4]
```diff
@@ -131,13 +132,17 @@ inline Configuration& Configuration::operator=(Configuration&& other) noexcept {
...
+  return *this;
 }
 
-inline void Configuration::clear() { m_appenders.clear(); }
+inline Configuration& Configuration::clear() {
+  m_appenders.clear();
+  return *this;
+}
 
 inline auto Configuration::getAppenders() const -> const AppenderList& {
```
Comment (andescu) [inline olp-cpp-sdk-core/include/olp/core/logging/Configuration.h]: This looks strange.
Change to
```cpp
inline const AppenderList& Configuration::getAppenders() const {
```

File: olp-cpp-sdk-core/include/olp/core/logging/Configuration.h (hunk 1 above)
Comment (andescu) [reply olp-cpp-sdk-core/include/olp/core/logging/Configuration.h]: Can we move using to top and use short doxygen like:

```cpp
class CORE_API Configuration {
 public:
/// Alias for a list of appenders
using AppenderList = std::vector<AppenderWithLogLevel>;
....
```

File: olp-cpp-sdk-core/include/olp/core/logging/ConsoleAppender.h [hunk 5]
```diff
@@ -49,7 +49,7 @@ class CORE_API ConsoleAppender : public IAppender {
    */
   inline void setMessageFormatter(MessageFormatter formatter);
```
Comment (andescu) [inline olp-cpp-sdk-core/include/olp/core/logging/ConsoleAppender.h]: This one?

File: olp-cpp-sdk-core/include/olp/core/logging/Configuration.h (hunk 2 above)
Comment (andescu) [reply olp-cpp-sdk-core/include/olp/core/logging/Configuration.h]: Please adapt to coding style
```cpp
AppenderList appenders_;
```

File: olp-cpp-sdk-core/include/olp/core/logging/Configuration.h (hunk 4 above)
Comment (andescu) [reply olp-cpp-sdk-core/include/olp/core/logging/Configuration.h]: Can we try to remove Configuration.cpp file?

File: olp-cpp-sdk-core/include/olp/core/logging/FilterGroup.h [hunk 6]
```diff
@@ -150,9 +150,15 @@ inline boost::optional<Level> FilterGroup::getLevel() const {
   return m_defaultLevel;
 }
 
-inline void FilterGroup::setLevel(Level level) { m_defaultLevel = level; }
+inline FilterGroup& FilterGroup::setLevel(Level level) {
```
Comment (andescu) [inline olp-cpp-sdk-core/include/olp/core/logging/FilterGroup.h]: Please move everything after here to FilterGroup.inl file

Comment (andescu) [reply olp-cpp-sdk-core/include/olp/core/logging/FilterGroup.h]: Align members to coding style
m_defaultLevel -> default_level_
Can we remove boost::optional somehow? Do we need it here? Can we not just say default level is trace?

File: olp-cpp-sdk-core/include/olp/core/logging/ConsoleAppender.h (hunk 5 above)
Comment (serhiikobalt) [reply olp-cpp-sdk-core/include/olp/core/logging/ConsoleAppender.h]: done

**Comment by serhiikobalt** [comment]
@andescu 
It's better if I'll do refactoring in a separate commit and not as part of olpedge-403

**Pull request merged**